### PR TITLE
Snake, try to remember some of the basics of CQC.

### PR DESCRIPTION
**Pull request by datlo**
Adds the CQC manual to nukie and tator uplinks at 9 TC, ported from TG.

CQC is a martial arts comparable to sleeping carp, however it doesn't let you deflect all projectiles. Instead it gives a 75% chance to block all melee attacks if you have throw intent on, effectively turning your throw hotkey into a block button. This protection does stack with a shield/an esword. 
It also boosts all of your basic unarmed attacks : Disarms inflict damage on top of a standard disarm, grabs are immediately aggressive, and basic harm inflicts extra damage.
Its combos are more focused on inflicting stuns and stamina damage into a knockout.

CQC does not restrict the usage of firearms in any way. The only downside is the cost of the 9 tc you spend on it. It can't be used with gloves of the north star.

CQC makes throwing bodies much easier because you instantly aggressive grab them. As we all know, this is the most robust ranged attack in the game.

Secret coderbus combo : Harm intent while resting lets you do a legsweep.

cqc

🆑
add: Added the CQC manual to nuke ops and traitor uplinks, letting them remember some of the basics of CQC.
/ 🆑

### CODE REVIEW

**Comment by ScratKnapp** [comment]
It seems like the last thing you'd be doing as a war or  regular assault ops would be dicking about with unarmed.

Though for Stealth Ops I could see this being friggin sweet.

**Comment by tigercat2000** [comment]
Why does this not include the ability for chefs to use it while they are inside the kitchen, as /tg/ has?

**Comment by KasparoVy** [comment]
Because perhaps the general impression was that idea would get insta shot down

**Comment by datlo** [comment]
Chefs were given the ability because TG greytide would always break into the kitchen. I don't feel this is an issue on paradise, and imo I don't think it fits the server. If maintainers want chefs to have access to it, tho, it should be a fairly simple copy paste job, just let me know.

**Comment by variableundefined** [comment]
I agree that giving chef CQC is not a good idea - They have no business fighting there. Kitchen break in seldomly happen here, too.

Leave that for nukies.

**Comment by granodd** [comment]
Chefs don't need CQC when they have Oven-Fu (the ability to smack intruders with roller pins, knives, food trays, not to mention slamming their faces into grills, fryers and ovens).

Otherwise I'm a little ambivalent on the CQC for nukies. I like it cos it gives more options for unique builds. On the other hand, I don't see it being utilized all that often, as it'd require a decent level of robustness to utilize to its maximum potential.

**Comment by TDSSS** [comment]
Could we potentially add this to normal traitor uplinks, too? Seems like it would be more useful for them than for nukies who usually go loud anyway.

**Comment by AzuleUtama** [comment]
> Could we potentially add this to normal traitor uplinks, too? Seems like it would be more useful for them than for nukies who usually go loud anyway.

Normal traitors have the carp scroll for that though. Nukies don't get one because it would be insanely overpowered if they combined that with a shielded hardsuit. CQC is a way for them to get the nice attack combos without the broken defence the scroll gives.

**Comment by datlo** [comment]
Important notes I forgot to mention and will edit in : CQC does not restrict the usage of firearms in any way. The only downside is the cost of the 8 tc you spend on it.

CQC makes throwing bodies much easier because you instantly aggressive grab them. As we all know, this is the most robust ranged attack in the game.

Secret coderbus combo : Disarm intent while resting lets you do a legsweep.

**Comment by KasparoVy** [comment]
Fuck yes prone attacka

Makes me wish we had crawling

File: code/modules/martial_arts/cqc.dm [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		return TRUE
+	if(findtext(streak,RESTRAIN_COMBO))
+		streak = ""
+		Restrain(A,D)
+		return TRUE
+	if(findtext(streak,PRESSURE_COMBO))
+		streak = ""
+		Pressure(A,D)
+		return TRUE
+	if(findtext(streak,CONSECUTIVE_COMBO))
```
Comment (farie82) [inline code/modules/martial_arts/cqc.dm]: Spacing in all the procs used here. example: Slam(A, D)

File: code/modules/martial_arts/cqc.dm [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+/datum/martial_art/cqc/proc/Consecutive(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	if(!D.stat)
+		D.visible_message("<span class='warning'>[A] strikes [D]'s abdomen, neck and back consecutively</span>", \
+							"<span class='userdanger'>[A] strikes your abdomen, neck and back consecutively!</span>")
+		playsound(get_turf(D), 'sound/weapons/cqchit2.ogg', 50, 1, -1)
+		var/obj/item/I = D.get_active_hand()
+		if(I &&  D.drop_item())
```
Comment (farie82) [inline code/modules/martial_arts/cqc.dm]: two spaces here between the && and D.drop_item

File: code/modules/martial_arts/cqc.dm [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+/datum/martial_art/cqc/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	add_to_streak("G",D)
+	if(check_streak(A,D))
+		return TRUE
+	var/obj/item/grab/G = D.grabbedby(A,1)
+	if(G)
+		G.state = GRAB_AGGRESSIVE //Instant aggressive grab
+	return TRUE
```
Comment (farie82) [inline code/modules/martial_arts/cqc.dm]: spacing

File: code/modules/martial_arts/cqc.dm [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+/datum/martial_art/cqc/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	add_to_streak("H",D)
+	if(check_streak(A,D))
```
Comment (farie82) [inline code/modules/martial_arts/cqc.dm]: spacing once more. And the line above

File: code/modules/martial_arts/cqc.dm [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+		D.apply_damage(10, BRUTE)
+		D.Weaken(3)
+	return TRUE
+
+/datum/martial_art/cqc/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	add_to_streak("D",D)
+	var/obj/item/I = null
+	if(check_streak(A,D))
```
Comment (farie82) [inline code/modules/martial_arts/cqc.dm]: space and 2 lines above

File: code/modules/martial_arts/martial.dm [hunk 6]
```diff
@@ -184,6 +189,24 @@
 	new /obj/effect/decal/cleanable/ash(get_turf(src))
 	qdel(src)
 
+/obj/item/CQC_manual
+	name = "old manual"
+	desc = "A small, black manual. There are drawn instructions of tactical hand-to-hand combat."
+	icon = 'icons/obj/library.dmi'
+	icon_state ="cqcmanual"
```
Comment (farie82) [inline code/modules/martial_arts/martial.dm]: space

**Comment by KasparoVy** [comment]
Spacing everywhere 🚀

File: code/modules/martial_arts/cqc.dm [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+	help_verb = /mob/living/carbon/human/proc/CQC_help
+	block_chance = 75
+
+/datum/martial_art/cqc/proc/drop_restraining()
+	restraining = FALSE
+
+/datum/martial_art/cqc/proc/check_streak(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	if(findtext(streak,SLAM_COMBO))
```
Comment (Aurorablade) [inline code/modules/martial_arts/cqc.dm]: As a rule, Always space between the comma and the next arg in the list..Even here.

File: code/modules/martial_arts/cqc.dm [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+	if(findtext(streak,PRESSURE_COMBO))
+		streak = ""
+		Pressure(A, D)
+		return TRUE
+	if(findtext(streak,CONSECUTIVE_COMBO))
+		streak = ""
+		Consecutive(A, D)
+	return FALSE
+
+/datum/martial_art/cqc/proc/Slam(mob/living/carbon/human/A, mob/living/carbon/human/D)
```
Comment (Aurorablade) [inline code/modules/martial_arts/cqc.dm]: on TG it does have logging, consider adding attack logs on the manuvers.

File: code/modules/martial_arts/cqc.dm [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+/datum/martial_art/cqc/proc/Consecutive(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	if(!D.stat)
+		D.visible_message("<span class='warning'>[A] strikes [D]'s abdomen, neck and back consecutively</span>", \
+							"<span class='userdanger'>[A] strikes your abdomen, neck and back consecutively!</span>")
+		playsound(get_turf(D), 'sound/weapons/cqchit2.ogg', 50, 1, -1)
+		var/obj/item/I = D.get_active_hand()
+		if(I && D.drop_item())
+			A.put_in_hands(I)
```
Comment (Aurorablade) [inline code/modules/martial_arts/cqc.dm]: Does this work more effectively then TGs version of dropping the item then putting it in hand?

Comment (Aurorablade) [reply code/modules/martial_arts/cqc.dm]: Rescending comment i just am compareing too many parts of this code at once.

File: code/modules/martial_arts/cqc.dm [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+	return TRUE
+
+/datum/martial_art/cqc/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	add_to_streak("H", D)
+	if(check_streak(A, D))
+		return TRUE
+	A.do_attack_animation(D)
+	var/picked_hit_type = pick("CQC'd", "Big Bossed")
```
Comment (Aurorablade) [inline code/modules/martial_arts/cqc.dm]: While fine on TG i am not sure this would fit well on our server.

Comment (KasparoVy) [reply code/modules/martial_arts/cqc.dm]: :(

File: code/modules/martial_arts/cqc.dm (hunk 8 above)
Comment (variableundefined) [reply code/modules/martial_arts/cqc.dm]: Please do

**Comment by variableundefined** [comment]
Raise the cost to 9 and add it as an option to traitor as requested on discord.

- Leaving this message here for transparency sake.

**Comment by datlo** [comment]
Added missing spacings
Added attack logging
Raised cost to 9 TC, made it tator and nukies, added to surplus, allowed
nukies to have gloves of north star, and updated description
Added a check preventing usage of CQC with gloves of north star
Fix a bug where attacks would not go through if a combo was done but the
target was critically injured (regular attacks now done instead)
Changed disarm logic, now always hits for 5 damage and use regular
disarm logic for disarming/pushing
Replaced "Big Bossed" attack verb by "neck chopped" and "gut punched"
because nofunallowed

**Comment by datlo** [comment]
this is ready for merge unless i missed something dumb (again)

**Comment by KasparoVy** [comment]
RIP Big Bossed

**Comment by Kyep** [comment]
I believe this PR is causing a runtime:
DEBUG: Runtime in items.dm,568: undefined proc or verb /mob/living/silicon/robot/check block().

**Pull request merged**